### PR TITLE
refs #3 Merge debugger support from qore-vscode-debug

### PR DESCRIPTION
**Pull request by tmandys**

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -93,6 +115,51 @@ export async function activate(context: vscode.ExtensionContext) {
...
+    /*context.subscriptions.push(vscode.debug.onDidChangeActiveDebugSession(session => {
+        console.log("extension.qore-debug-vscode.onDidChangeActiveDebugSession(session:" + JSON.stringify(session) + ")");
+    }));*/
+    /*context.subscriptions.push(vscode.debug.onDidReceiveDebugSessionCustomEvent(event => {
+        console.log("extension.qore-debug-vscode.onDidReceiveDebugSessionCustomEvent(event:" + JSON.stringify(event) + ")");
+    }));*/
+    // loaded scripts
+    //vscode.window.registerTreeDataProvider('extension.qore-debug-vscode.loadedScriptsExplorer', new loadedScripts_1.LoadedScriptsProvider(context))
+    //context.subscriptions.push(vscode.commands.registerCommand('extension.qore-debug-vscode.pickLoadedScript', loadedScripts_1.pickLoadedScript));
+    //context.subscriptions.push(vscode.commands.registerCommand('extension.qore-debug-vscode.openScript', (session, source) => loadedScripts_1.openS
```
Comment (sejvlond) [inline src/extension.ts]: :153-:162 are these lines needed, or why are they commented out?

File: package.json [hunk 2]
```diff
@@ -2,14 +2,14 @@
     "name": "qore-vscode",
     "displayName": "Qore",
     "description": "Qore language support for VS Code",
-    "version": "0.3.3",
+    "version": "0.3.4",
```
Comment (sejvlond) [inline package.json]: don't we want to increase more then just a patch version? we are forcing user to upgrade qore, I would go with at least minor, maybe even call it version `1.0.0`?

File: src/extension.ts [hunk 3]
```diff
@@ -4,11 +4,35 @@ import * as child_process from 'child_process';
...
+}
+
+// tutorial abouut "=>" https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions
 export async function activate(context: vscode.ExtensionContext) {
     console.log('Activating qore-vscode extension');
 
     let qore_executable: string = vscode.workspace.getConfiguration("qore").get("executable") || "qore";
+    console.log('Qore executable: '+qore_executable);
 
     // Find out if Qore and the astparser module are present.
```
Comment (sejvlond) [inline src/extension.ts:106]: maybe we should add a check here, whether the qore bin is the requested version?

Comment (Foxhoundn) [reply src/extension.ts:106]: It's a little thing, but `let` should only be used if you plan on re-declaring the variable. If the variable does not change it should be `const`.

**Review by Foxhoundn — APPROVED** [review]
Seems aight.

File: package.json (hunk 2 above)
Comment (omusil24) [reply package.json]: That's right. The version should be updated to at least 0.4, but maybe in this case 1.0.0 is a good idea. Also, please add a note about this to `CHANGELOG.md` file, in the same style as the other entries.

Comment (omusil24) [reply package.json]: Actually just noticed that there is already an entry in the changelog. But the version should be increased to 0.4.

**Comment by tmandys** [comment]
Review changes applied

File: src/extension.ts [hunk 4]
```diff
@@ -5,10 +5,24 @@ import * as vscode from 'vscode';
...
 
-var qoreExecutable: string;
-var debugAdapter: string;
+let qoreExecutable: string;
+let debugAdapter: string;
+/*
+    We need list of programs in GUI via registerCommand(extension.qore-vscode.getProgram)
+    but the connection is also resolvable in in registerCommand(extension.qore-vscode.getConnection).
+    getConnection is called first but does not updates config passed to getProgram. So as
+    workqround we pass value via global variable currentConnect which won't work if any
```
Comment (omusil24) [inline src/extension.ts]: Typos: `in in` -> `in`, `updates` -> `update`, `workqround` -> `workaround`

File: src/extension.ts [hunk 5]
```diff
@@ -269,17 +299,18 @@ function getExecutableArguments(configuration: DebugConfiguration): string[] {
             args.push(configuration.respTimeout);
         }
         if (typeof configuration.headers !== "undefined") {
-            for (var _hdr of configuration.headers) {
+            for (let _hdr of configuration.headers) {
                 if (typeof _hdr.name !== "string" || typeof _hdr.value !== "string") {
-                    throw new Error("Wrong name or value for a header in: "+JSON.stringify(_hdr));
+                    let hdrs: string = JSON.stringify(_hdr);
+                    throw new Error(t`"Wrong name or value for a header in: ${hdrs}`);
```
Comment (omusil24) [inline src/extension.ts]: Left-over `"`

**Review by martin-zemek — APPROVED** [review]

**Pull request merged**